core/src/test/java/org/elasticsearch/index/mapper/DocumentMapperMergeTests.java

```diff
@@ -296,4 +298,24 @@ public void testMergeAddingParent() throws IOException {
         Exception e = expectThrows(IllegalArgumentException.class, () -> initMapper.merge(updatedMapper.mapping(), false));
         assertThat(e.getMessage(), containsString("The _parent field's type option can't be changed: [null]->[parent]"));
     }
+
+    public void testMergeDate() throws IOException {
```

Comment:
Can you move this test to `DateFieldMapperTests` instead and undo changes to this file?